src/static/src/components/Table.js

```diff
@@ -0,0 +1,83 @@
+import toTimeString from "../utils/toTimeString.js";
+import { PlayIcon } from "./icons/index.js";
+
+const Table = ({ jojoDoc, setCursor }) => {
+  const { useState, useEffect } = preact;
+  const [editElement, setEditElement] = useState();
+  const text = jojoDoc.segments;
+
+  useEffect(() => {
+    let el = editElement;
+    if (!el) return () => {};
+
+    const id = el.dataset.id;
+    const listener = () => {
+      const inputedText = el.querySelector("[contenteditable]").innerText;
+      const textRef = text.find((t) => t.id == id);
+      if (textRef.text !== inputedText) {
+        textRef.text = inputedText;
+      }
+    };
+
+    el.classList.toggle("selected");
+    el.addEventListener("focusout", listener, false);
+
+    return () => {
+      el.classList.remove("selected");
+      el.removeEventListener("focusout", listener);
+    };
+  }, [editElement]);
+
+  useEffect(() => {
+    const transcriptionTable = document.getElementById("transcription");
+    const clickListener = (event) => {
+      const el = event.target;
+      const tr = el.closest("tr");
+      const playBtn = el.closest("svg");
```

Comment:
This will need some more logic to work than this, but should we do this?
```suggestion
      const playBtn = el.closest("svg");
      if(!audio){
      	alert("you need to upload a soundfile to listen)
      	}
      }
```